src/rt/aaA.d

```diff
@@ -152,6 +152,21 @@ private:
         deleted = 0;
         GC.free(obuckets.ptr); // safe to free b/c impossible to reference
     }
+
+    void clear() pure nothrow
+    {
+        foreach (ref b; buckets[firstUsed .. $])
+        {
+            if (b.filled)
+            {
+                b.hash = HASH_DELETED;
+                b.entry = null;
+            }
+        }
+        deleted = used;
+        firstUsed = cast(uint) dim;
+        resize(INIT_NUM_BUCKETS);
```

Comment:
Note, it would be nice to be able to specify such behaviors for the implementation.